### PR TITLE
Substrate ibc 0.9.43

### PR DESCRIPTION
**Pull request by baidang201**
https://github.com/ggxchain/planning/issues/155

### CODE REVIEW

File: runtime/brooklyn/src/chain_extensions.rs [hunk 1]
```diff
@@ -479,7 +507,19 @@ where
...
+			trace!(
+					target: "runtime",
+					"root origin not supported"
+			);
+			// TODO: expand XvmErrors with BadOrigin
+			return Err(DispatchError::Other(
+				"ChainExtension root origin not supported",
+			));
+		}
+	};
```
Comment (obsessed-cake) [inline runtime/brooklyn/src/chain_extensions.rs:522]: looks like, we can create a function for this and use one 3 times.

File: runtime/brooklyn/src/lib.rs [hunk 2]
```diff
@@ -863,6 +920,42 @@ impl_runtime_apis! {
...
+			let (weight_limit, proof_size_base_cost) =
+				match <Runtime as pallet_evm::Config>::GasWeightMapping::gas_to_weight(
+					gas_limit,
+					without_base_extrinsic_weight
+				) {
+					weight_limit if weight_limit.proof_size() > 0 => {
+						(Some(weight_limit), Some(estimated_transaction_len as u64))
+					}
+					_ => (None, None),
+				};
```
Comment (obsessed-cake) [inline runtime/brooklyn/src/lib.rs:957]: looks like, we can create a function for this and use one 2 times.

File: runtime/sydney/src/lib.rs [hunk 3]
```diff
@@ -746,6 +802,42 @@ impl_runtime_apis! {
...
+			let (weight_limit, proof_size_base_cost) =
+				match <Runtime as pallet_evm::Config>::GasWeightMapping::gas_to_weight(
+					gas_limit,
+					without_base_extrinsic_weight
+				) {
+					weight_limit if weight_limit.proof_size() > 0 => {
+						(Some(weight_limit), Some(estimated_transaction_len as u64))
+					}
+					_ => (None, None),
+				};
```
Comment (obsessed-cake) [inline runtime/sydney/src/lib.rs:839]: looks like, we can create a function for this and use one 2 times.

File: runtime/brooklyn/src/lib.rs (hunk 2 above)
Comment (baidang201) [reply runtime/brooklyn/src/lib.rs:957]: i find we do not need create a function. there are too many input/output parameters if make a function.
there are look the same with astar.

File: runtime/sydney/src/lib.rs (hunk 3 above)
Comment (baidang201) [reply runtime/sydney/src/lib.rs:839]: i find we do not need create a function. there are too many input/output parameters if make a function.
there are look the same with astar.

**Review by akorchyn — APPROVED** [review]

**Review by obsessed-cake — APPROVED** [review]

**Comment by akorchyn** [comment]
I think we can merge it. @baidang201

**Pull request merged**